agent_build_refactored/managed_packages/managed_packages_builders.py

```diff
@@ -972,269 +1082,477 @@ def create_build_dependencies_step(
+            tracked_files_globs=[
+                "agent_build_refactored/managed_packages/scalyr_agent_python3/agent-python3-config",
+                "agent_build_refactored/managed_packages/scalyr_agent_python3/install-scriptlets/postinstall.sh",
+                "agent_build_refactored/managed_packages/scalyr_agent_python3/install-scriptlets/preuninstall.sh"
+            ],
+            base=PREPARE_TOOLSET_STEPS[Architecture.X86_64],
+            required_steps={
+                "BUILD_OPENSSL_1_1_1": BUILD_OPENSSL_STEPS[OPENSSL_VERSION_TYPE_1_1_1][architecture],
+                "BUILD_OPENSSL_3": BUILD_OPENSSL_STEPS[OPENSSL_VERSION_TYPE_3][architecture],
+                "BUILD_PYTHON_WITH_OPENSSL_1_1_1": BUILD_PYTHON_STEPS[OPENSSL_VERSION_TYPE_1_1_1][architecture],
+                "BUILD_PYTHON_WITH_OPENSSL_3": BUILD_PYTHON_STEPS[OPENSSL_VERSION_TYPE_3][architecture],
+            },
+            environment_variables={
+                "PYTHON_SHORT_VERSION": EMBEDDED_PYTHON_SHORT_VERSION,
+                "INSTALL_PREFIX": "/opt/scalyr-agent-2-dependencies",
+                "LIBSSL_DIR": libssl_dir
+            },
+            github_actions_settings=GitHubActionsSettings(
+                cacheable=True
+            )
         )
-    )
 
-# Step that prepares initial environment for X86_64 build environment.
-INSTALL_GCC_7_GLIBC_ARM64 = EnvironmentRunnerStep(
-        name="install_gcc_7_arm64",
-        script_path="agent_build_refactored/managed_packages/steps/install_gcc_7_arm64.sh",
-        base=DockerImageSpec(
-            name="centos:7",
-            platform=DockerPlatform.ARM64.value
-        ),
-        github_actions_settings=GitHubActionsSettings(
-            cacheable=True,
-            run_in_remote_docker=True
+        steps[architecture] = step
+
+    return steps
+
+
+def create_build_dev_requirements_steps() -> Dict[Architecture, ArtifactRunnerStep]:
+    """
+    Create steps that build all agent project requirements.
+    """
+    steps = {}
+    for architecture in SUPPORTED_ARCHITECTURES:
+        run_in_remote_docker = architecture != Architecture.X86_64
+
+        build_dev_requirements_step = ArtifactRunnerStep(
+            name=f"build_dev_requirements_{architecture.value}",
+            script_path="agent_build_refactored/managed_packages/steps/build_dev_requirements.sh",
+            tracked_files_globs=[
+                "dev-requirements-new.txt",
+            ],
+            base=INSTALL_BUILD_ENVIRONMENT_STEPS[architecture],
+            required_steps={
+                "BUILD_PYTHON_DEPENDENCIES": BUILD_PYTHON_DEPENDENCIES_STEPS[architecture],
+                "BUILD_OPENSSL": BUILD_OPENSSL_STEPS[DEFAULT_OPENSSL_VERSION][architecture],
+                "BUILD_PYTHON": BUILD_PYTHON_STEPS[DEFAULT_OPENSSL_VERSION][architecture],
+            },
+            environment_variables={
+                "RUST_VERSION": "1.63.0",
```

Comment:
Would it make sense to put this value in some module level constant to make it easier to see it at a glance with other versions + update it?